wled00/wled.cpp

```diff
@@ -697,7 +701,48 @@ bool WLED::initEthernet()
 #else
   return false; // Ethernet not enabled for build
 #endif
+}
+
+// performs asynchronous scan for available networks (which may take couple of seconds to finish)
+// returns true if only one wifi is configured or scan completed
+bool WLED::findWiFi(bool doScan) {
+  if (multiWiFi.size() <= 1) {
+    DEBUG_PRINTLN(F("Defaulf WiFi used."));
+    selectedWiFi = 0;
+    return true;
+  }
+
+  if (doScan) WiFi.scanDelete();  // restart scan
+
+  int status = WiFi.scanComplete();
 
+  if (status == WIFI_SCAN_FAILED) {
+    DEBUG_PRINTLN(F("WiFi scan started."));
+    WiFi.scanNetworks(true);  // start scanning in asynchronous mode
+    return false;
+  }
+  if (status >= 0) { // status contains number of found networks
+    DEBUG_PRINT(F("WiFi scan completed: ")); DEBUG_PRINTLN(status);
+    int rssi = -9999;
+    for (int o = 0; o < status; o++) {
+      DEBUG_PRINT(F(" WiFi available: ")); DEBUG_PRINT(WiFi.SSID(o));
+      DEBUG_PRINT(F(" RSSI: ")); DEBUG_PRINT(WiFi.RSSI(o)); DEBUG_PRINTLN(F("dB"));
+      for (unsigned n = 0; n < multiWiFi.size(); n++)
+        if (!strcmp(WiFi.SSID(o).c_str(), multiWiFi[n].clientSSID)) {
+          // find the WiFi with the strongest signal (but keep priority of entry if signal difference is not big)
+          if ((n < selectedWiFi && WiFi.RSSI(o) > rssi-10) || WiFi.RSSI(o) > rssi) {
```

Comment:
By default, `WiFi.scanNetworks()` (either Async or no) will always "return" the networks already ordered from nearest to farthest. There's no need to sort them again loosing CPU cycles.

Reply:
There is no sorting involved. By nearest you are referring to strongest signal I assume.
Even though the strongest signal may be, in your opinion, the best choice I have a use case where it is not, so the order of entered WiFis in settings takes precedence *if the difference in signal strength is low (10dB)*.